paasta_tools/cli/cmds/spark_run.py

```diff
@@ -606,6 +636,30 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     return spark_app_name
 
 
+def _calculate_docker_memory_cpu_limit(spark_conf):
+    try:
+
+        docker_memory_limit_str = spark_conf.get(
+            "spark.driver.memory", DEFAULT_DRIVER_MEMORY_BY_SPARK
+        )
+        match = re.match(r"([0-9]+)([a-z]*)", docker_memory_limit_str)
```

Comment:
should the memory unit party of the regex be the subset of allowed units? e.g., 
```Python
        match = re.match(r"([0-9]+)(mb|gb)?", docker_memory_limit_str.lower())
```

or do we have a pre-commit hook or something that ensures that people don't make mistakes here?

Reply:
If the units are wrong, the spark driver will fail and/or so does creating the container. 
I think adding this check will be an overkill here, and would make it harder to debug in future and needs to consider all possible edge cases (e.g. 1000 is valid value too just like 1000m).